### DISCUSSION TITLE
Why can't I bind command with parameter by CommunityToolkit.Mvvm?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#12299] [Q&A] [answered]
Here is XAML:
```
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"	
			 xmlns:Local="using:AvaloniaApplication1"
             x:Class="AvaloniaApplication1.MainView" FontFamily="avares://AvaloniaApplication1/#Source Han Sans VF">
	  <Button Command="{Binding InputCommand}" CommandParameter="{Binding $self}">Test</Button>
</UserControl>
```

And here is code-behind:
```
using Avalonia;
using Avalonia.Controls;
using Avalonia.Markup.Xaml;
using CommunityToolkit.Mvvm.Input;
using System.Collections.Generic;

namespace AvaloniaApplication1;

public partial class MainView : UserControl
{    
    public MainView()
    {
        InitializeComponent();
        this.DataContext = this;
        
    }
    [RelayCommand]
    public void InputCommand(Button button)
    { 

    }  
}
```

After the program ran, the button is always disable. When I open the devTools will find the command of the button is null.

What's wrong with my code?

**Comment by lukaspirkl** [comment, marked as answer]
It is because when you use the `[RelayCommand]` the property is generated with the name of your method and the postfix "Command" is added to it. And this generated property is the one you should bind to. So you should change the ViewModel code to:
```
[RelayCommand]
public void Input(Button button)
{ 

}  
```

**Reply by kyuranger** [reply]
All right! It works! Thank you!

**Comment by rabbitism** [comment]
If you want to use CTMVVM generated command, then the command name is wrong. If you want to bind to method directly, in Avalonia it only supports one parameter with type as `object`.

**Reply by kyuranger** [reply]
Firstly, as you said, the problem of command name is right. 

Secondary, I just tested the parameter type. It works with the parameter type of origin control, for example:
```
[RelayCommand]
public void Input(Button button)
{ 

}  
```

 But it does not work with TemplatedControls, for example:
```
[RelayCommand]
public void Input(TemplatedControl1 button)
{ 

}  
```

**Reply by timunie** [reply]
RelayCommand is for ViewModel, not for the View. In views, I'd rather listen for Click event.

**Reply by kyuranger** [reply]
Thanks for your suggestion.
